atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1038,6 +1004,31 @@ private HeldLocksToken getExpiredHeldLocksToken() {
                 "Dummy thread");
     }
 
+    private long concurrentlyIncrementValueThousandTimesAndGet() throws InterruptedException, ExecutionException {
+        CompletionService<Void> executor = new ExecutorCompletionService<Void>(
+                PTExecutors.newFixedThreadPool(8));
+        final Cell cell = Cell.create(PtBytes.toBytes("row1"), PtBytes.toBytes("column1"));
+        Transaction t1 = txManager.createNewTransaction();
+        t1.put(TABLE, ImmutableMap.of(cell, EncodingUtils.encodeVarLong(0L)));
+        t1.commit();
+        for (int i = 0; i < 1000; i++) {
+            executor.submit(() -> {
+                txManager.runTaskWithRetry((TxTask) t -> {
+                    long prev = EncodingUtils.decodeVarLong(
+                            t.get(TABLE, ImmutableSet.of(cell)).values().iterator().next());
+                    t.put(TABLE, ImmutableMap.of(cell, EncodingUtils.encodeVarLong(prev + 1)));
+                    return null;
+                });
+                return null;
+            });
+        }
+        for (int i = 0; i < 1000; i++) {
+            Future<Void> future = executor.take();
+            future.get();
+        }
+        t1 = txManager.createNewTransaction();
+        return EncodingUtils.decodeVarLong(t1.get(TABLE, ImmutableSet.of(cell)).values().iterator().next());
+    }
```

Comment:
**NON-ACTIONABLE**: The above could do with a significant cleanup, but given that it was just extracted from the bunch of duplicated methods and lives in test code I'm okay with living with this for now.